docs-src/guides-multi-tenancy.md

```diff
@@ -0,0 +1,37 @@
+External Secrets Operator provides different modes of operation to fulfill ogranizational needs. This guide outlines the flexibility of ESO and should give you a first impression of how you can employ this operator in your organization.
+
+For a multi-tenant deployment you should first examine your organizational structure:
+
+1. what roles (i.e. *Application Developers*, *Cluster Admins*, ...) do you have in your organization,
+2. what responsibilities do they have and
+3. how does that map to Kubernetes RBAC roles.
+
+Further, you should examine how your external API provider manages access control for your secrets. Can you limit access by secret names (e.g. `db/dev/*`)? Or only on a bucket level? Please keep in mind that not all external APIs provide fine-grained access management for secrets.
+
+**Note:** The following examples should **not** be considered as best practice but rather as a example to show how to combine different mechanics and techniques for tenant isolation.
+
+### Shared ClusterSecretStore
+
+![Shared CSS](./pictures/diagrams-multi-tenancy-shared.png)
+
+A Cluster Administrator deploys a `ClusterSecretStore` (CSS) and manages access to the external API. The CSS is shared by all tenants within the cluster. Application Developers do reference it in a `ExternalSecret` but can not create a ClusterSecretStores or SecretStores on their own. Now all application developers have access to all the secrets. You probably want to limit access to certain keys or prefixes that should be used.  ESO provides a very simple mechanic to limit access to certain keys on a namespace basis by using annotations on the `Kind=Namespace` resource. More advanced validation should be done with an Admission Webhook, e.g. with [Kyverno](https://kyverno.io/) or [Open Policy Agent](https://www.openpolicyagent.org/)).
```

Comment:
Nit: Markdown lines should be <120 characters (doesn't need to be strict character limit, just roughly somewhere around there). This is to make diffs clearer, otherwise when making a typo change, the entire line (in this case 720 characters) is shown in the diff. 

No need to fix all instances up in this case, but for future reference as we continue to enhance the documentation.